fe/fe-core/src/main/java/com/starrocks/server/SharedDataStorageVolumeMgr.java

```diff
@@ -0,0 +1,124 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.server;
+
+import com.google.common.base.Preconditions;
+import com.staros.proto.FileStoreInfo;
+import com.staros.util.LockCloseable;
+import com.starrocks.common.AlreadyExistsException;
+import com.starrocks.common.AnalysisException;
+import com.starrocks.common.DdlException;
+import com.starrocks.storagevolume.StorageVolume;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class SharedDataStorageVolumeMgr extends StorageVolumeMgr {
+    @Override
+    public Long createStorageVolume(String name, String svType, List<String> locations, Map<String, String> params,
+                                    Optional<Boolean> enabled, String comment)
+            throws AlreadyExistsException, AnalysisException, DdlException {
+        if (exists(name)) {
+            throw new AlreadyExistsException(String.format("Storage volume '%s' already exists", name));
+        }
+        StorageVolume sv = new StorageVolume(0, name, svType, locations, params, enabled.orElse(true), comment);
+        return Long.valueOf(GlobalStateMgr.getCurrentState().getStarOSAgent().addFileStore(sv.toFileStoreInfo()));
+    }
+
+    @Override
+    public void removeStorageVolume(String name) throws AnalysisException, DdlException {
+        StorageVolume sv = getStorageVolumeByName(name);
+        Preconditions.checkState(sv != null,
```

Comment:
shall this protected by lock? otherwise during the removal, someone set the to-be-removed-volume as default or bind it to db/table?